File: POO/Heritage/todo-sequence2/cowboy.cpp
#include "cowboy.h"

// Constructeurs et Destructeur
Cowboy::Cowboy(const string nom/*=""*/, const string boissonFavorite/*="whisky"*/, int popularite/*=0*/, string adjectif/*="vaillant"*/)\
: Humain(nom, boissonFavorite), popularite(popularite), adjectif(adjectif)
{

}

Cowboy::~Cowboy()
{

}

string Cowboy::getAdjectif() const
{
    return adjectif;
}

void Cowboy::sePresente() const
{
    cout << "(" << nom << ") -- " << "Bonjour, je suis le " << this->getAdjectif()<< " " << this->getNom() << " et j’aime le " << this->getBoissonFavorite() << endl;
}   

File: POO/Java/sequence4/Cowboy.java
public class Cowboy extends Humain
{
    private String nom;
    private String boissonFavorite = "whisky";
    private int popularite = 0;
    private String adjectif = "vaillant";

    public Cowboy(String nom)
    {
        super();
        this.nom = nom;
    }

    public void sePresente()
    {
        String texte = "";
        texte = "Bonjour, je suis le " + getAdjectif() + getNom() + " et j’aime le " + getBoissonFavorite();
        parle(texte);
    }

    public String getAdjectif()
    {
        return adjectif;
    }
}